app/models/customer_export.rb

```diff
@@ -28,24 +29,63 @@ def export_to_salesforce(pagination_cursor: nil)
     end
   end
 
-  def synchronize_square_customer(customer)
+  private
+
+  def create_square_customer(customer)
     square_customer = SquareCustomer.find_or_initialize_by(square_id: customer.id)
     if square_customer.new_record?
-      push_customer_to_salesforce(customer)
+      id = find_square_customer_in_salesforce(customer)
+      if id.nil? && @wet_run
+        id = create_square_customer_in_salesforce(customer)
+      elsif id.nil?
+        Rails.logger.info "Dry run: would create #{customer} in Salesforce"
+      end
 
-      square_customer.pushed_as_of = customer.updated_at
-      square_customer.save!
+      if !id.nil?
+        square_customer.salesforce_id = id
+        square_customer.save!
+      end
     end
   end
 
-  def push_customer_to_salesforce(customer)
-    salesforce_client.synchronize_contact(
-      Salesforce::Contact.new(
+  def find_square_customer_in_salesforce(customer)
+    # Match on first name, last name, and email
+    if customer.given_name.present? && customer.family_name.present? && customer.email_address.present?
```

Comment:
Instead of 3 different checks, would it make more sense to build a hash based on the presence of these fields and just send that in?